### PR TITLE
Fix calculations in `model2netcdf.ED2()` on E files

### PR DESCRIPTION
**Pull request by Aariq**
## Description

This is a near total re-write of the `read_E_files()` function that extracts data from monthly .h5 files output by ED2.  The re-write simplifies code by extracting variables with the same dimensions into data frames and joining them before doing any calculations or unit conversions.  This simplifies the logic greatly (IMO).

It also removes DDBH (cm/plant/yr) as one of the variables extracted from the -E- files, since I'm no longer sure it even makes sense to summarize across cohorts.

The time dimensions were getting calculated incorrectly for monthly -E- file data as well (#3069), and that is now fixed in this PR.

## Motivation and Context

Previously, the conversions were operating on the false assumption that variables like NPP and transpiration were coming out of ED2 in per-area units (https://github.com/EDmodel/ED2/pull/384).  Also, there was previously no weighting by patch area when summing across cohorts and patches. See more in #3126 

## Review Time Estimate

- [ ] Immediately
- [x] Within one week
- [ ] When possible

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue) 
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My change requires a change to the documentation.
- [x] My name is in the list of CITATION.cff
- [x] I have updated the CHANGELOG.md.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: models/ed/R/model2netcdf.ED2.R [hunk 1]
```diff
@@ -971,100 +970,115 @@ read_E_files <- function(yr, yfiles, h5_files, outdir, start_date, end_date,
...
+      ),
+      #(or a mean when it makes sense)
+      DBH_mean = mean(.data$DBH),
+      .groups = "drop"
+    )
+  
+  # Rename variables to PEcAn standard and convert to list
+  out_list <- out %>% 
+    dplyr::select(
+      #PEcAN name    #ED2 name
```
Comment (dlebauer) [inline models/ed/R/model2netcdf.ED2.R:1073]:
```suggestion
      #PEcAn name    #ED2 name          
```

**Review by dlebauer — APPROVED** [review]

**Comment by Aariq** [comment]
Just installed this on Welsch and getting this warning for every variable, so something's not right still:

```
With 'AGB_PFT': [1] "ncvar_put: warning: you asked to write 10 values, but the passed data array has 24 entries!" 
With 'BSEEDS': [1] "ncvar_put: warning: you asked to write 10 values, but the passed data array has 24 entries!" 
```

Going back to a draft for now.

File: models/ed/R/model2netcdf.ED2.R [hunk 2]
```diff
@@ -971,100 +970,115 @@ read_E_files <- function(yr, yfiles, h5_files, outdir, start_date, end_date,
...
+  out_list <- out %>% 
+    dplyr::select(
+      #PEcAn name    #ED2 name          
+      "AGB_PFT"    = "AGB_CO",
+      "BSEEDS"     = "BSEEDS_CO",
+      "DBH"        = "DBH_mean",
+      "NPP_PFT"    = "MMEAN_NPPDAILY_CO",
+      "TRANSP_PFT" = "MMEAN_TRANSP_CO",
+      "DENS"       = "NPLANT"
+    ) %>% as.list()
```
Comment (Aariq) [inline models/ed/R/model2netcdf.ED2.R:1080]: I think the elements of this list need to be arrays with PFT as one of the dimensions.  Need to check with `put_E_values()` to see what it's expecting.

**Comment by Aariq** [comment]
Whooops, I guess `read_E_files()` has to read only one year at a time for it to be compatible with `put_E_values()`

**Pull request merged**